src/OpenTelemetry/BaseExporter.cs

```diff
@@ -42,6 +42,16 @@ public abstract class BaseExporter<T> : IDisposable
     /// <returns>Result of the export operation.</returns>
     public abstract ExportResult Export(in Batch<T> batch);
 
+    /// <summary>
+    /// Exports a batch of telemetry objects.
+    /// </summary>
+    /// <param name="batch">Batch of telemetry objects to export.</param>
+    /// <returns>Result of the export operation.</returns>
+    public virtual Task<ExportResult> ExportAsync(Batch<T> batch)
```

Comment:
A related discussion on this topic we had in OTel Rust community : https://github.com/open-telemetry/opentelemetry-rust/pull/2027
OTel Rust only has async API for the exporters. This is causing perf issues when the exporter does not benefit from async API (like etw exporters which has no async need as it don't talk to disk/network etc.)

We are also considering adding both options in OTel Rust, with one calling the other automatically!
Just shared as these are similar, though in diff. languages!

Reply:
The allocation overhead for calling `ExportAsync` instead of `Export` should just be the `Task` allocation. AFAIK, state machines are kept on the stack while the methods return synchronously. If a method needs to yield the execution because of a blocking operation (e.g. reading a socket), then the state machine will be moved to the heap.

In this [short snippet](https://sharplab.io/#v2:D4AQDABCCMCsDcBYAUCkBOCBBAzgTwDsBjAeQDcBTAJwGVCiAKASiVTZADZt9jzq7izFAG8UEcRAD0kiABMA9hBzyAthQAuACwCWBAOZiJIAOxQOAOgDCqgA4AbDRVmdWAXyA===) we can see that the state machine is a struct and when the async method doesn't return synchronously (i.e. in the `if (!awaiter.IsCompleted)` branch), `AwaitUnsafeOnCompleted` is called which [boxes the state machine](https://github.com/dotnet/runtime/blob/8ae3796ef109962725833610c26e3f0bb06e2988/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs#L92).

So I'm not very concerned about the performance impact for synchronous exporters, especially that the method should only be called every X seconds.

One thing that could be done is to use `ValueTask` to avoid the `Task` allocation. It is used for example in [Stream.ReadAsync(Memory<byte>)](https://github.com/dotnet/runtime/blob/8ae3796ef109962725833610c26e3f0bb06e2988/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L308). Though `ValueTask` has more constraints than `Task` so it can be a tough decision to make. That's why I always use `Task` and it never was a bottleneck. More reading from our savior Stephen Toub: https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask.

Reply:
> So I'm not very concerned about the performance impact for synchronous exporters, especially that the method should only be called every X seconds.

The export method is called for every Log, every Span when using sync exporters like ETW. (they don't do any batching)

Reply:
That could make sense to use ValueTask in that case. Could you give me a pointer to the ETW exporter. I can't seem to find it

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Exporter.Geneva/Internal/ReentrantExportProcessor.cs - This is the exporting processor used with ETW exporters.